#!/usr/bin/env python
PACKAGE = "hector_motion_detection"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

# Generic Parameters
gen.add("motion_detect_downscale_factor", double_t, 0, "Image Downscale Factor", 2, 1, 10)
gen.add("motion_detect_inv_sensivity", double_t, 0, "Motion Inverse Sensivity", 40, 1, 500)
gen.add("motion_detect_use_initial_flow", bool_t, 0, "Use Initial Flow", False)
gen.add("motion_detect_image_flow_filter", bool_t, 0, "Use Image Flow Filtering", False)
gen.add("motion_detect_threshold", int_t, 0, "Motion Detect Threshold", 20, 0, 255)
gen.add("motion_detect_min_area", double_t, 0, "Min Motion Area", 200, 1, 100000)
gen.add("motion_detect_min_blob_dist", double_t, 0, "Min Motion Blob Distance", 50, 1, 1000)
gen.add("motion_detect_dilation_size", int_t, 0, "Dilation Size", 3, 1, 300)
gen.add("motion_detect_flow_history_size", int_t, 0, "History Size", 3, 0, 100)
gen.add("percept_class_id", str_t, 0, "The percept class string", "victim")

exit(gen.generate(PACKAGE, "motion_detection", "MotionDetection"))
